### PR TITLE
add scraper package

### PR DESCRIPTION
**Pull request by nicoxx**
-only reddit is parsed, using PRAW library
-add common package: shared objects used by learner
-remove test projects from pylint analysis
-use VCRPY for integration tests of scraper
-add setup.py to make packages importable and solve py.test issues

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -4,18 +4,18 @@ python:
...
-  -nv
+- wget http://storage.googleapis.com/appengine-sdks/featured/google_appengine_1.9.23.zip -nv
 - unzip -q google_appengine_1.9.23.zip -d lib
 - ./lib/google_appengine/dev_appserver.py ./src/server/server/ --skip_sdk_update_check &
 - sleep 3
 script:
 - py.test src
 - coverage run --source=src --omit=*tests* -m py.test src
-- pylint src/server/server src/server/tests src/functests
+- pylint src/server/server src/scraper/scraper/ src/common/common
```
Comment (nicoxx) [inline .travis.yml]: linter on test projects might be overkill and could reduce productivity

File: pylintrc [hunk 2]
```diff
@@ -240,7 +240,8 @@ generated-members=REQUEST,acl_users,aq_parent
 [FORMAT]
 
 # Maximum number of characters on a single line.
-max-line-length=100
+max-line-length=125
```
Comment (nicoxx) [inline pylintrc]: 125 is the standard of what is 'diffable' on github, default 100 is too restrictive

File: pylintrc [hunk 3]
```diff
@@ -352,7 +353,7 @@ exclude-protected=_asdict,_fields,_replace,_source,_make
 [DESIGN]
 
 # Maximum number of arguments for function / method
-max-args=5
+max-args=7
```
Comment (nicoxx) [inline pylintrc]: pylint by default is very restrictive...

File: requirements.txt [hunk 4]
```diff
@@ -0,0 +1,2 @@
+jsonpickle
```
Comment (nicoxx) [inline requirements.txt]: jsonpickle for serialization to json, praw is reddit API

File: src/common/setup.py [hunk 5]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+ 
+from distutils.core import setup
+
+setup(
+    name='common',
+    packages=['common']
+)
```
Comment (nicoxx) [inline src/common/setup.py]: setup.py at root is mandatory to run 'pip install -e' and then import between different packages

File: src/scraper/tests/test_reddit.py [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    def test_is_valid_submission_with_over_18_return_false(self):
+        submission = MagicMock(is_self=False, over_18=True)
+        self.assertFalse(reddit._is_valid_submission(submission))
+
+    def test_is_valid_submission_ok_return_false(self):
+        submission = MagicMock(is_self=False, over_18=False)
+        self.assertTrue(reddit._is_valid_submission(submission))
+
+if __name__ == '__main__':
+    unittest.main()
```
Comment (nicoxx) [inline src/scraper/tests/test_reddit.py]: I can't create a praw.Submission object, I mock it

File: src/scraper/tests/test_scraper.py [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+    # to record:
+        # set disconnected=False (to set 2sec delay between requests)
+        # delete test_run_scraper.yaml
+        # set record_mode='once'
+    def test_run_scraper(self):
+        nb_doc = 20  # 20 to keep test under 2sec
+        curr_doc = 0
+
+        directory = os.path.dirname(os.path.abspath(__file__))
```
Comment (nicoxx) [inline src/scraper/tests/test_scraper.py]: integration test for scraper, use vcrpy (register then replay html requests to not make tests reliables)

File: tools/run_pylint.py [hunk 8]
```diff
@@ -50,7 +50,9 @@ def construct_command():
         command.append('--ignore=%s' % ','.join(IGNORE_PATTERNS))
     if ADDITIONAL_PARAMETERS:
         command.extend(ADDITIONAL_PARAMETERS)'''
-    command = ['pylint', 'src/server/server', 'src/server/tests', 'src/functests', '-f', 'parseable']
+    command = ['pylint',
+    'src/server/server', 'src/scraper/scraper', 'src/common/common',
+     '-f', 'parseable']
```
Comment (nicoxx) [inline tools/run_pylint.py]: equivalent to pylint.rc for pycharm

File: .travis.yml (hunk 1 above)
Comment (m-mokhtari) [inline .travis.yml]: I don't see any folder "scraper/scraper" and "common/common", am I missing something ?

Comment (m-mokhtari) [inline .travis.yml]: Forgot this comment :).

**Pull request merged**